feat: In-page navigation for the About page

dzole0311 opened the pull request:
Related issue: https://github.com/NASA-IMPACT/veda-ui/issues/1634

Opening a PR as I need some feedback / thoughts on the behavior and implementation (see "Current issues" below)

Few observations:

1. We were previously using a USWDS `SideNav` on the About page, which is not the same as the official `InPageNavigation` component from USWDS, which is what the design uses
2. Directly using the `InPageNavigation` component from react-uswds had issues with our SSR About page. I ended up rebuilding it manually using USWDS in-page nav classes (code-wise, it's not a huge change)
3. The in-page navigation doesn't work well on shorter pages like our About page, where sections are often just one sentence. According to USWDS guidance, in-page nav isn't necessary on pages that require little or no scrolling

Re: point 3, I additionally tested the USWDS demo [hosted here](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/iframe.html?id=components-in-page-navigation--default&args=&viewMode=story), by replacing its content with smaller paragraphs; It also shows quirky behavior when sections are small or closely spaced:

https://github.com/user-attachments/assets/5d2edc25-bbca-4704-a98e-86c355be524c

Changes:

1. Replace `SideNav` with `InPageNavigation`
2. Tweak the rootMargin and thresholds for the `IntersectionObserver` in the `useActiveSection` hook
3. Update labels/titles for the in-page navigation items to match the designs

Current issues:

1. The last 1-2 sections, especially when viewed on a big screen, can't be reached easily since they are closely spaced (or their content is too small). There are some workarounds that I tried, but none that worked reliably
~~2. The focus outline is not persisted when clicking on an item in the `InPageNav`~~

aboydnw commented:
I think our goal with this should be a) stability and b) polish.

So for example I'm less worried about this issue:

> 1. The last 1-2 sections, especially when viewed on a big screen, can't be reached easily since they are closely spaced (or their content is too small). There are some workarounds that I tried, but none that worked reliably

Than I am about this issue:

> 2. The focus outline is not persisted when clicking on an item in the InPageNav

Because I think the second one is more noticeable and reflects poorly on the site. I don't think this interaction needs to be perfect, just somewhat polished and reliable.

dzole0311 commented:
Thanks @vgeorge for the help with the scroller!

> * Considering side-nav.scss has styles unrelated to the side-nav itself, I suggest renaming it to layout.scss

The styles were about the side-nav, just scoped with an `.about` selector. I switched to the `layout.scss` now + renamed `SideNav` to `InPageNav` to better reflect which USWDS component we use.

> * The focus highlight on click makes the navigation a bit confusing as there might be two different highlighted sections. I would be in favor of not highlighting it as mouse click + keyboard nav looks like a rare use case

I agree, but it seems to be the default USWDS behavior, [example here](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/iframe.html?id=components-in-page-navigation--default&args=&viewMode=story#aliquam-erat-volutpat-velit-vitae-ligula-volutpat):

1. Click on one of the nav items
2. Scroll up/down -> the outline remains even though a different item is outlined

I suggest we match the default USWDS behavior and change/tweak only if design thinks otherwise.

vgeorge reviewed (approved):
@dzole0311 looks good to merge 👍
